scripts/gen_dynamic_openapi.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -e
+
+# Generate an OpenAPI document for all backends.
+#
+# Assumptions:
+#
+#   1. Vault has been checked out at an appropriate version and built
+#   2. vault executable is in your path
+#   3. Vault isn't already running
+
+echo "Starting Vault..."
+if pgrep -x "vault" > /dev/null
+then
+    echo "Vault is already running. Aborting."
+    exit 1
+fi
+
+vault server -dev -dev-root-token-id=root &
+sleep 2
+VAULT_PID=$!
+
+echo "Mounting all builtin backends..."
+
+# Read auth backends
+flag=false
+re='".*"'
+while read p; do
+    if [[ $p == *"credentialBackends:"* ]] ; then
+        flag=true
+    elif [ $flag = true ] && [[ $p = *"}"* ]]  ; then
+        break
+    elif [ $flag = true ] && [[ $p =~ $re ]]; then
+        backend=${BASH_REMATCH[0]}
+        var1=$(sed -e 's/^"//' -e 's/"$//' <<<"$backend") 
```

Comment:
Is this line going to exclude the `// deprecated` entries. It may be worthwhile to do that in the overall regex.

Reply:
I actually haven't added the deprecation logic yet. My plan was to run it through a separate regex and not include those matches. B/c from what I can tell, to include this in the current regex would require something that isn't currently supported in bash regex. B/c we'd be attempting to negate the word, which I can't seem to find an easy way to do. But I could be missing something.

Reply:
This line just removes the quotes that get matched by the regex

Reply:
I added some deprecated logic that just checks if the line contains "deprecated"